client/wfprev-war/src/main/angular/src/app/components/edit-project/fiscal-map/fiscal-map.component.ts

```diff
@@ -0,0 +1,231 @@
+              .pipe(
+                map(boundary => boundary ? ({
+                  activityGuid: activity.activityGuid,
+                  fiscalYear: activity.fiscalYear,
+                  boundary: boundary?._embedded?.activityBoundary
+                }) : null),
+              )
+          );
+  
+          forkJoin(boundaryRequests).subscribe((allResults) => {
+            this.allActivityBoundaries = allResults.filter(r => 
+              r !== null && r.boundary && Object.keys(r.boundary).length > 0
+            );
+
+            if (this.allActivityBoundaries.length > 0 && this.map) {
+              this.plotBoundariesOnMap(this.allActivityBoundaries)
+            } else{
+              this.getProjectCoordinates();
+            }
+          });
+        });
+      });
+    }
+  }
+  
+  plotBoundariesOnMap(boundaries: any[]): void {
+    const currentFiscalPolygons: L.Layer[] = [];
+  
+    boundaries.forEach(boundaryEntry => {
+      const fiscalYear = boundaryEntry.fiscalYear;
+      let color = '';
+  
+      if (fiscalYear < this.currentFiscalYear) {
+        color = this.fiscalColorMap.past;
+      } else if (fiscalYear === this.currentFiscalYear) {
+        color = this.fiscalColorMap.present;
+      } else {
+        color = this.fiscalColorMap.future;
+      }
+  
+      boundaryEntry.boundary.forEach((item: any) => {
+        const geometry = item.geometry;
+        if (!geometry) return;
+  
+        const geoJsonOptions: L.GeoJSONOptions = {
+          style: {
+            color,
+            weight: 2,
+            fillOpacity: 0.1
+          }
+        };
+  
+        const addToMap = (geom: any) => {
+          const layer = L.geoJSON(geom, geoJsonOptions).addTo(this.map!);
+          if (fiscalYear === this.currentFiscalYear) {
+            currentFiscalPolygons.push(layer);
+          }
+        };
+  
+        if (geometry.type === 'GeometryCollection') {
+          geometry.geometries.forEach((subGeom: any) => addToMap(subGeom));
+        } else {
+          addToMap(geometry);
+        }
+      });
+    });
+  
+    // Zoom to current fiscal year polygons
+    if (currentFiscalPolygons.length > 0) {
+      const group = L.featureGroup(currentFiscalPolygons);
+      this.map!.fitBounds(group.getBounds(), { padding: [20, 20] });
+    }
+  }
+  
+
+  initMap(): void {
+    const mapContainer = document.getElementById('fiscalMap');
+    if (mapContainer && (mapContainer as any)._leaflet_id != null) {
+      (mapContainer as any)._leaflet_id = null;
+    }
+
+    this.map = L.map('fiscalMap', {
+      center: [49.00005, -124.0001], // Center of the boundary
+      zoom: 14,
+      zoomControl: true,
+    });
+
+    (this.map.zoomControl as L.Control.Zoom).setPosition('topright');
+
+    L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+      attribution: '© OpenStreetMap contributors'
+    }).addTo(this.map);
+
+    this.addLegend();
+  }
+
+  addLegend(): void {
```

Comment:
Can this function be put in a common file or service so it can be used on the map and project and details pages?

Reply:
added to tools.ts